### DISCUSSION TITLE
Index duplicate files?

### DISCUSSION THREAD

**Discussion by nitobuendia** [photoprism/photoprism#3450] [Q&A] [answered]
## Situation

I am using Photoprism in read only and I use folders to manage my photos. I use folders because it's backed up from a few sources and I do want to use folders instead of other features (e.g. albums).

I have a folder structure like this:

* `/photoprism/originals/<year>/<month>/`
* `/photoprism/originals/albums/<album-name>/`

My photos tend to belong to both folders. One for when they were shot (`<year>/<month>`) and one (or more) for the albums (`<album-name>`) to which they belong.

While de-duplication is usually a good feature. In my case, I want to make sure that photos appear in both the date and any albums where they belong. However, since photoprism detects the albums photos as duplicates, it doesn't add them.

## Question

Is there a way to show the duplicated photos on the right folders? Alternatively, less desirable, is there a way disable the skip for duplicated photos and make sure that they show in both folders?

Thanks!

## Version Details

Photoprism version: `photoprism version 230603-378d4746a-Linux-AMD64-Plus` - installed via docker compose on an Ubuntu server machine.

**Comment by lastzero** [comment, marked as answer]
> Is there a way to show the duplicated photos on the right folders?

PhotoPrism keeps duplicates in a separate index. So it seems to be possible to implement this. However, this is the first time anyone has asked for such a feature, and we already have a long list of other feature requests, so it may take us a while to get to it. You could, of course, fork our code and implement it for your own needs.

**Comment by nitobuendia** [comment]
Fair enough. Appreciated for taking the time to reply, @lastzero . Any pointers?

~~From the code, I can see that both [files.go](https://github.com/photoprism/photoprism/blob/378d4746ad69c8d321f1e89b5a51f99c4a4b8d60/internal/photoprism/files.go#L42) and [photos.go](https://github.com/photoprism/photoprism/blob/378d4746ad69c8d321f1e89b5a51f99c4a4b8d60/internal/photoprism/photos.go#L39) are calling [IndexedFiles](https://github.com/photoprism/photoprism/blob/378d4746ad69c8d321f1e89b5a51f99c4a4b8d60/internal/query/files.go#L177) which retrieves both `duplicates` and `files`. Additionally,  from what I can see from the database, I have the duplicates which has the `file_hash` and the `file_name`. This should allow me to "merge" the indeces and get all the metadata needed.~~

~~However, I am not sure which file/method is responsible for reading the photo map/list and showing them on `/library/browse` and `/library/folders`.~~

Nevermind, I missed the obvious `/frontend/` folder.

**Comment by rheckly** [comment]
I just started to use PhotoPrism a month ago and I'm quite happy.
I have all my pictures (originals) in folders which are like albums.
So I mapped this folder as my originals and they are now indexed.

Now I figured out, that the way I handled / organized my pictures is not compatable with PhotoPrism. Indeed I have copies of pictures in different folders - also my SD Card Backup is in this "original" folders.
Now I know there is a specific picture, but when I look in PhotoPrism I only find it in my SD Backup Folder - and not in the original folder.

Therefore I'd like to have the ability to have duplcute pictures in my full database of photoprism. 
Either the Originals or Folders should show the real content / physical contant of a folder (which is the event in my library) - independent if the pic is already in another folder.

otherwise it's irritating when I look into a folder and only 10 pics of physically 50 pics in the folder are shown in photoprism.

A setting to disable the duplicate function would be a starting.
A way to show in folders the real physical pictures would be very helpfull.

I now go and rename my SD Backup Folder not to be Indexed (take them to the exception List).
